dna/models.py

```diff
@@ -814,6 +815,93 @@ def predict_regression(self, data, *, verbose=False):
 
         return predictions
 
+class LinearRegressionBaseline(RegressionModelBase):
+    def __init__(self, seed=0):
+        RegressionModelBase.__init__(self, seed=seed)
+        self.regressor = linear_model.LinearRegression()
+        self.fitted = False
+        self.pipeline_key = 'pipeline'
+        self.steps_key = 'steps'
+        self.prim_name_key = 'name'
+
+    def fit(self, data, *, validation_data=None, output_dir=None, verbose=False):
+        self.fitted = True
+        self.one_hot_primitives_map = self._one_hot_encode_mapping(data + validation_data)
```

Comment:
You cannot use the validation data to make the one-hot encoding mapping. The train data should be used to make the mapping, then the mapping should be applied to both the train and validation data.

Reply:
That assumes that our random split of the dataset and primitives contains all primitives (likely it does, but it's not guaranteed).  If you're sure I'll change it but it might throw an error based on the split.  It's not like doing this causes any data leakage.

Reply:
All these meta-models are trying to learn the behavior of the primitives to be able to make predictions about them. If there are any primitives in the test/validation data that are not found in the training data, we have no hope of making any predictions about them. Having an error in the one-hot encoding would help us see that we have out-of-train-set primitives.

Reply:
Okay sounds good, I'll remove it.  I'm worried it won't work on our small dataset, but that's not a huge loss.